api/src/main/java/org/pmiops/workbench/exceptions/ExceptionAdvice.java

```diff
@@ -6,13 +6,23 @@
 import org.pmiops.workbench.model.ErrorResponse;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.bind.annotation.ResponseStatus;
 
 
 @ControllerAdvice
 public class ExceptionAdvice {
+  private static final Logger log = Logger.getLogger(ExceptionAdvice.class.getName());
+
+  @ExceptionHandler({HttpMessageNotReadableException.class})
+  public ResponseEntity<?> messageNotReadableError(Exception e) {
+    log.log(Level.INFO, "failed to parse HTTP request message, returning 400", e);
+    return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(
+        WorkbenchException.errorResponse("failed to parse valid JSON request message")
```

Comment:
I think you can use our BadRequestException here? https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/exceptions/BadRequestException.java

Reply:
It's possible that would work, but the code that translates our exceptions into responses (including BadRequestException) is 2 lines below this method. These handlers are logically being applied at the same level so even if it worked today I'm not sure I want to depend on that working in future versions of Spring.